ALICE3/TableProducer/onTheFlyTracker.cxx

```diff
@@ -0,0 +1,159 @@
+#include "Framework/runDataProcessing.h"
+#include "Framework/RunningWorkflowInfo.h"
+#include "Framework/HistogramRegistry.h"
+#include "Framework/O2DatabasePDGPlugin.h"
+#include "Framework/ASoAHelpers.h"
+#include "Common/DataModel/TrackSelectionTables.h"
+#include "Common/Core/trackUtilities.h"
+#include "ReconstructionDataFormats/DCA.h"
+#include "DetectorsBase/Propagator.h"
+#include "DetectorsBase/GeometryManager.h"
+#include "CommonUtils/NameConf.h"
+#include "CCDB/CcdbApi.h"
+#include "CCDB/BasicCCDBManager.h"
+#include "DataFormatsParameters/GRPMagField.h"
+#include "DataFormatsCalibration/MeanVertexObject.h"
+#include "CommonConstants/GeomConstants.h"
+
+using namespace o2;
+using namespace o2::framework;
+
+struct OnTheFlyTracker {
+  Produces<aod::Collisions> collisions;
+  Produces<aod::McCollisionLabels> collLabels;
+  Produces<aod::StoredTracks> tracksPar;
+  Produces<aod::TracksExtension> tracksParExtension;
+  Produces<aod::StoredTracksCov> tracksParCov;
+  Produces<aod::TracksCovExtension> tracksParCovExtension;
+  Produces<aod::McTrackLabels> tracksLabels;
+  Produces<aod::TracksDCA> tracksDCA;
+
+  bool fillTracksDCA = false;
+
+  // necessary for particle charges
+  Service<O2DatabasePDG> pdg;
+
+  o2::base::Propagator::MatCorrType matCorr = o2::base::Propagator::MatCorrType::USEMatCorrNONE;
+
+  void init(o2::framework::InitContext& initContext)
+  {
+    // Checking if the tables are requested in the workflow and enabling them
+    auto& workflows = initContext.services().get<RunningWorkflowInfo const>();
+    for (DeviceSpec const& device : workflows.devices) {
+      for (auto const& input : device.inputs) {
+        if (input.matcher.binding == "TracksDCA") {
+          fillTracksDCA = true;
+        }
+      }
+    }
+  }
+
+  /// Function to convert a McParticle into a perfect Track
+  /// \param particle the particle to convert (mcParticle)
+  /// \param o2track the address of the resulting TrackParCov
+  template <typename McParticleType>
+  void convertMCParticleToO2Track(McParticleType& particle, o2::track::TrackParCov& o2track)
+  {
+    auto pdgInfo = pdg->GetParticle(particle.pdgCode());
+    int charge = 0;
+    if (pdgInfo != nullptr) {
+      charge = pdgInfo->Charge();
+    }
+    std::array<float, 5> params;
+    std::array<float, 15> covm = {0.};
+    float s, c, x;
+    o2::math_utils::sincos(particle.phi(), s, c);
+    o2::math_utils::rotateZInv(particle.vx(), particle.vy(), x, params[0], s, c);
+    params[1] = particle.vz();
+    params[2] = 0.; // since alpha = phi
+    auto theta = 2. * std::atan(std::exp(-particle.eta()));
+    params[3] = 1. / std::tan(theta);
+    params[4] = charge / particle.pt();
+
+    // Initialize TrackParCov in-place
+    new (&o2track)(o2::track::TrackParCov)(x, particle.phi(), params, covm);
+  }
+
+  template <typename CollType, typename TTrackPar>
+  void FillTracksPar(CollType& coll, aod::track::TrackTypeEnum trackType, TTrackPar& trackPar)
```

Comment:
```suggestion
  template <typename CollType, typename TTrackPar>
  void fillTracksPar(CollType& coll, aod::track::TrackTypeEnum trackType, TTrackPar& trackPar)
```
Please add at least one line of documentation above.

Reply:
fixed!